include/mbgl/style/conversion/heatmap_color_property_value.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <mbgl/style/heatmap_color_property_value.hpp>
+#include <mbgl/style/conversion.hpp>
+#include <mbgl/style/conversion/constant.hpp>
+#include <mbgl/style/conversion/function.hpp>
+#include <mbgl/style/conversion/expression.hpp>
+#include <mbgl/style/expression/value.hpp>
+#include <mbgl/style/expression/is_constant.hpp>
+#include <mbgl/style/expression/is_expression.hpp>
+#include <mbgl/style/expression/find_zoom_curve.hpp>
+
+namespace mbgl {
+namespace style {
+namespace conversion {
+
+template <>
+struct Converter<HeatmapColorPropertyValue> {
+    optional<HeatmapColorPropertyValue> operator()(const Convertible& value, Error& error) const {
+        if (isUndefined(value)) {
+            return HeatmapColorPropertyValue();
+        } else if (isExpression(value)) {
+            optional<std::unique_ptr<Expression>> expression = convert<std::unique_ptr<Expression>>(value, error, expression::type::Color);
```

Comment:
Can the `unique_ptr` in this optional be ever be empty? If so, we should check for that to avoid null dereferences further down.

Reply:
It should never be null (we use the optional to represent parsing failure).  It's a bug rather than a recoverable failure if the pointer is null, so if we want to check here, it should be with an `assert`